core/src/main/scala/kafka/log/Log.scala

```diff
@@ -444,11 +444,10 @@ class Log(@volatile var dir: File,
    *
    * @param records The log records to append
    * @param assignOffsets Should the log assign offsets to this message set or blindly apply what it is given
-   * @param leaderEpochCache Optional cache of Leader Epoch Offsets.
    * @throws KafkaStorageException If the append fails due to an I/O error.
    * @return Information about the appended messages including the first and last offset.
    */
-  def append(records: MemoryRecords, assignOffsets: Boolean = true, leaderEpochCache: LeaderEpochCache = leaderEpochCache): LogAppendInfo = {
+  def append(records: MemoryRecords, assignOffsets: Boolean = true, leaderEpoch: Int = leaderEpochCache.latestEpoch()): LogAppendInfo = {
```

Comment:
Could we add leaderEpoch to the comment above and mention that it's only used when assignOffsets is true? Also, could we remove the default value for leaderEpoch and make it the second param? Currently, it seems that at least StressTestLog and TestLinearWriteSpeed will fail since the default leaderEpoch is -1.